### PR TITLE
dev/financial#148 Simplify parameters passed to completeOrder

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Simplifies the code in Event batch update to no longer interact with the BaseIPN class

Before
----------------------------------------
$baseIPN validate called to retrieve 'objects' - but in practice doing the same as contribution->fetch() plus an incorrect guess at the relevant payment processor

After
----------------------------------------
just calls contribution->fetch()

Technical Details
----------------------------------------
Now that completeOrder has had most interaction with objects var cleaned up
we know only 2 objects are actually used
- contribution - which we can fetch in-function
- paymentProcessor - which we were actually only guessing

This alters it to only load & pass contribution. The value of passing in
processor seems lost in time. This code can only be reached for pay_later contributions - see

CRM_Contribute_BAO_Contribution::checkOnlinePendingContribution
and specifically
https://github.com/civicrm/civicrm-core/blob/318c63045e82d30ed7fe8cdacb91a6d31fc5331b/CRM/Contribute/BAO/Contribution.php#L2033

and only when submitted via an online event (same function) so by definition there IS NO PAYMENT PROCESSOR that applies.

Given that we can just do the minimum to load the contribution and pass that in.

Testing notes
----------------------------------------
I added CRM_Event_Form_Task_BatchTest::testSubmit prior to starting cleanup on this code. It passes through the altered lines & should lock in 'no change of behaviour'

### CODE REVIEW

File: CRM/Event/Form/Task/Batch.php [hunk 1]
```diff
@@ -387,10 +359,10 @@ public static function updateContributionStatus($params) {
     // @todo use the api - ie civicrm_api3('Contribution', 'completetransaction', $input);
     // as this method is not preferred / supported.
     CRM_Contribute_BAO_Contribution::completeOrder($input, [
-      'related_contact' => $ids['related_contact'] ?? NULL,
```
Comment (eileenmcnaughton) [inline CRM/Event/Form/Task/Batch.php:390]: We don't support onBehalf for event registrations

**Comment by eileenmcnaughton** [comment]
@seamuslee001 this has passed now - I'm trying to eliminate all the places that call 

->validateData

**Comment by mattwire** [comment]
Ok, this could use an `r-run` from the event batch form

**Comment by eileenmcnaughton** [comment]
Closed by accident as I deleted the branch :-)
Re-opened here https://github.com/civicrm/civicrm-core/pull/18479

**Pull request closed without merging**